src/js/utils/generate_filename.js

```diff
@@ -12,11 +12,11 @@ function zeroPad(value, width) {
     return valuePadded;
 }
 
-export function generateFilename(prefix, suffix) {
+export function generateFilename(prefix, suffix, boardName) {
     const date = new Date();
     let filename = prefix;
 
-    if (FC.CONFIG) {
+    if (FC.CONFIG && !boardName) {
```

Comment:
Perhaps we could add FC.CONFIG.boardName to this block and move it down to the line 41